wrappers/flutter/lib/src/infra/common/common_fitatu_scanner_preview.dart

```diff
@@ -9,77 +10,174 @@ import '../../scanner_preview_mixin.dart';
+    switch (state) {
+      case AppLifecycleState.resumed:
+        resumeFromBackground = false;
+        startScanner();
+        break;
+      case AppLifecycleState.paused:
+        resumeFromBackground = true;
+        break;
+      case AppLifecycleState.inactive:
+        if (!resumeFromBackground) {
+          stopScanner();
+        }
+        break;
+      case AppLifecycleState.detached:
+        break;
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
-    return MobileScanner(
-      controller: controller,
-      onScannerStarted: (_) async => await controller.resetZoomScale(),
-      onDetect: (response) {
-        final List<Barcode> nonEmptyBarcodes = response.barcodes.where((barcode) => barcode.rawValue != null).toList();
-        for (final barcode in nonEmptyBarcodes) {
-          widget.onSuccess(barcode.rawValue!);
-          return;
-        }
-      },
-      errorBuilder: (_, exception, __) {
-        _handleError(exception);
+    final arguments = mobileScannerArguments;
+    return LayoutBuilder(builder: (context, constraints) {
+      final scanWindowSize = constraints.maxHeight * widget.options.cropPercent;
+      final scanWindow = Rect.fromCenter(
+        center: Offset(constraints.maxWidth / 2, constraints.maxHeight / 2),
+        width: scanWindowSize,
+        height: scanWindowSize,
+      );
+
+      return Stack(
+        fit: StackFit.expand,
+        children: [
+          MobileScanner(
+            controller: controller,
+            scanWindow: scanWindow,
+            onDetect: (response) {
+              final barcodes = response.barcodes.where((b) => b.rawValue != null).whereType<Barcode>().toList();
+              if (barcodes.isEmpty) {
+                widget.onResult(null);
+              } else {
+                widget.onResult(barcodes.first.rawValue);
+              }
+            },
+          ),
+          if (arguments != null)
+            Builder(
+              builder: (context) {
+                final metrix = CameraPreviewMetrix(
+                  cropRect: scanWindow,
+                  width: constraints.maxWidth,
+                  height: constraints.maxHeight,
+                  rotationDegrees: 90,
+                );
 
-        return const CommonScannerError();
-      },
-    );
+                return widget.overlayBuilder?.call(context, metrix) ?? PreviewOverlay(cameraPreviewMetrix: metrix);
+              },
+            ),
+        ],
+      );
+    });
   }
 
   @override
-  void setTorchEnabled({required bool isEnabled}) => controller.toggleTorch();
+  Future<void> setTorchEnabled({required bool isEnabled}) async {
+    try {
+      await controller.toggleTorch();
+    } on Exception catch (e) {
+      setException(e);
+    } finally {
+      safeSetState();
+    }
+  }
 
   @override
   bool isTorchEnabled() => controller.torchState.value == TorchState.on;
+
+  Future<void> startScanner() async {
```

Comment:
o widzę że przepisałeś uruchamianie IOS, to chyba nie ten task?

Reply:
To twoje rozwiązanie wywalało się właśnie tam gdzie mówiłem. Wykrywanie błędów jest możliwe tylko jak sami startujemy skaner, bo możemy łapać exception

Reply:
a w jaki sposób łapiesz pozostałe exceptiony  (np. przy włączaniu/wyłączaniu latarki?)

Reply:
Wszystko co odwołuje się do controllera jest w try catch